students/DiannaTingg/lessons/lesson04/assignment/basic_operations.py

```diff
@@ -0,0 +1,140 @@
+"""
+Basic operations for the customer database.
+"""
+
+import logging
+import csv
+import peewee as pw
+import customer_model as cm
+import create_customers as cc
+
+# pylint: disable-msg=line-too-long
+# pylint: disable-msg=too-many-arguments
+# pylint: disable-msg=no-value-for-parameter
+
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+FORMATTER = logging.Formatter(LOG_FORMAT)
+
+FILE_HANDLER = logging.FileHandler('db.log')
+FILE_HANDLER.setFormatter(FORMATTER)
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(FILE_HANDLER)
+
+LOGGER.info("Defining basic operations for the customer database.")
+
+
+def add_customer(customer_id, first_name, last_name, home_address, phone_number, email_address, status, credit_limit):
+    """
+    Adds a new customer to the database.
+    """
+    try:
+        with cm.DATABASE.transaction():
+            new_customer = cm.Customer.create(
+                customer_id=customer_id,
+                first_name=first_name,
+                last_name=last_name,
+                home_address=home_address,
+                phone_number=phone_number,
+                email_address=email_address,
+                status=status,
+                credit_limit=credit_limit)
+            new_customer.save()
+            LOGGER.info("Successfully added Customer %s: %s %s to the database.", customer_id, first_name, last_name)
+    except pw.IntegrityError:
+        LOGGER.error("Error adding %s %s. Customer id %s already exists in the database.",
+                     first_name, last_name, customer_id)
+        raise pw.IntegrityError
+
+
+def upload_csv(filename):
+    """
+    Adds customer data from a csv file.
+    """
+    with open(filename, newline="") as csvfile:
+        all_customers = csv.reader(csvfile)
+        headers = next(all_customers, None)  # Skip the header row
+
+        with cm.DATABASE.atomic():
+            cm.Customer.insert_many(all_customers, headers).execute()
+
+
+def search_customer(customer_id):
+    """
+    Returns a dictionary with customer's first name, last name, email address, and phone number.
+    If no customer is found, returns an empty dictionary.
+    """
+    try:
+        current_customer = cm.Customer.get(cm.Customer.customer_id == customer_id)
+
+        customer_keys = ["first_name", "last_name", "email_address", "phone_number"]
+
+        customer_values = [current_customer.first_name, current_customer.last_name,
+                           current_customer.email_address, current_customer.phone_number]
+
+        customer_dict = dict(zip(customer_keys, customer_values))
+
+        LOGGER.info("Found customer %s: %s %s.", customer_id, current_customer.first_name, current_customer.last_name)
+
+        return customer_dict
```

Comment:
Nice. It is technically faster, as well as more in line with what we are learning, to do a dictionary comprehension, rather than the dict/zip combination. Also, this is too much for a try block. Think about which line is going to give you the does not exist exception, and only have that line in the try block.